x-pack/platform/plugins/shared/index_management/__jest__/client_integration/helpers/setup_environment.tsx

```diff
@@ -75,14 +75,18 @@ applicationService.capabilities = {
   },
 };
 
+const http = httpServiceMock.createSetupContract();
+
+http.get.mockResolvedValue({ count: '10000' });
+
```

Comment:
Global mock affects every GET request in the testbed. That’s risky because it can mask wrong URLs/params, break unrelated GET expectations, and it also returns `count` as a string vs the runtime contract `{ count: number }`.

Suggestion: 

wire the new doc-count mock into the existing httpRequestsMockHelpers (in `__jest__/client_integration/helpers/http_requests.ts`) by adding something like `setGetIndexDocCountResponse(indexName, count)` that mocks only `GET ${INTERNAL_API_BASE_PATH}/index_doc_count/{indexName}`. Then remove the global `http.get` override from `setup_environment.tsx` and ensure the app uses the same `httpSetup` instance `(core.http)` that `httpRequestsMockHelpers` controls. This keeps request mocking scoped, consistent, and avoids accidental side effects across tests.